app/src/main/java/com/android/streetworkapp/model/event/EventViewModel.kt

```diff
@@ -15,10 +23,29 @@ open class EventViewModel(private val repository: EventRepository) : ViewModel()
     return repository.getNewEid()
   }
 
+  /** Fetch all events from the database. */
+  fun getEvents() {
+    repository.getEvents(
+        onSuccess = {
+          if (it.isEmpty()) {
+            _uiState.value = EventOverviewUiState.Empty
+          } else {
+            _uiState.value = EventOverviewUiState.NotEmpty(it)
+          }
+        },
+        onFailure = { Log.e("FirestoreError", "Error getting events: ${it.message}") })
+  }
+
   /**
    * Add a new event to the database.
    *
    * @param event The event to add.
    */
   fun addEvent(event: Event) = viewModelScope.launch { repository.addEvent(event) }
 }
+
+sealed class EventOverviewUiState {
+  data class NotEmpty(val eventList: List<Event>) : EventOverviewUiState()
+
+  data object Empty : EventOverviewUiState()
+}
```

Comment:
(second part of the previous comment): As mentioned in the last comment, I am not sure that using this class has an advantage over using just the List <Event>, is there a reason to keep it?